### PR TITLE
Framework: media modal janitorial

### PR DESCRIPTION
**Pull request by retrofox**
It's a PR with janitorial purpose.

- re-write extending from Component class
- use { localize } helper
- replace replaceState by setState
- fix eslint warnings
- etc.

The most significant change among all of these is adding a tidy component named `<FilterItem />` used for avoiding to bind a function to the component property.

### Testing

Play add/remove media files from different places such as site icon, post editor, featured image, etc. Just in case I leave some links

* Since adding a new image or set the featured image: `http://calypso.localhost:3000/post/<your-testing-blog>`

* Site icon: `http://calypso.localhost:3000/settings/general/<your-testing-blog>`

Everything should be 🆗

### CODE REVIEW

File: client/my-sites/media-library/filter-bar.jsx [hunk 1]
```diff
@@ -1,110 +1,100 @@
...
+	};
 
-		switch ( this.props.filter ) {
+	getSearchPlaceholderText() {
+		const { filter, translate } = this.props;
+		switch ( filter ) {
 			case 'images':
-				placeholderText = this.translate( 'Search images…', { textOnly: true } );
-				break;
+				return translate( 'Search images…', { textOnly: true } );
```
Comment (retrofox) [inline client/my-sites/media-library/filter-bar.jsx]: @yoavf If I'm not wrong I saw a comment of you talking about that textOnly isn't needed. is it correct?

Comment (yoavf) [inline client/my-sites/media-library/filter-bar.jsx]: Yes, it's no longer needed.

**Comment by retrofox** [comment]
Having trouble with the tests here. It's like shallow isn't working in the same way when the Component is built using `class`. Asked here: https://github.com/airbnb/enzyme/issues/208#issuecomment-278050889

**Comment by gwwar** [comment]
@retrofox, the test issue stems from this test setup. Your janitorial wraps the class in localize, so this is no longer available.
```jsx
EditorMediaModal.prototype.translate = i18n.translate;
```

File: client/post-editor/media-modal/index.jsx [hunk 2]
```diff
@@ -161,35 +173,35 @@ export const EditorMediaModal = React.createClass( {
...
+		recordEvent( 'Upload Media' );
+	}
 
-	onAddAndEditImage: function() {
+	onAddAndEditImage = () => {
 		MediaActions.setLibrarySelectedItems( this.props.site.ID, [] );
 
 		this.props.setView( ModalViews.IMAGE_EDITOR );
-	},
+	}
```
Comment (gwwar) [inline client/post-editor/media-modal/index.jsx]: I suppose these need to be bound for the accept lib? Following our style with static props, could we add `;` here for these functions. This is an assignment statement.

**Comment by gwwar** [comment]
Changes make sense here. @hoverduck or @alisterscott, would it be possible to run e2e tests against this branch? The media modal is just so easy to regress.

File: client/post-editor/media-modal/index.jsx (hunk 2 above)
Comment (retrofox) [inline client/post-editor/media-modal/index.jsx]: Updating ... thanks

**Comment by hoverduck** [comment]
I ran the media upload and post editor tests (which include media as well), everything came back clean

**Comment by gwwar** [comment]
Thanks @hoverduck !

**Comment by retrofox** [comment]
Thanks folks for the review. Shipping now.

**Pull request merged**